<a href="https://colab.research.google.com/github/mjiii25/posco-academy/blob/main/Big-Data/Battery-Project/hypothesis-test.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **가설 검정**

In [ ]:
from google.colab import drive
drive.mount('/content/mydrive')

Mounted at /content/mydrive


In [162]:
import pandas as pd
import numpy as np

import pandas as pd
import scipy.stats as stats
from scipy.stats import mannwhitneyu

import matplotlib.pyplot as plt
import matplotlib

from statsmodels.formula.api import ols
from statsmodels.stats.anova import anova_lm

### **데이터 구성하기**

In [163]:
df_raw_process = pd.read_csv('/content/mydrive/MyDrive/posco-big-data/bat_process.csv', encoding = 'euc-kr')


print(df_raw_process.shape)
df_raw_process.head()

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (97,101,102,103) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


(43754, 109)


lot_id    tray_id                dt_start     cell_id judge  rta1_cell_no  \
0  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10747    양품            10   
1  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10748    양품             7   
2  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10749    양품            19   
3  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10750    양품            24   
4  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10751    양품            17   

  rta1_box_col rta1_box_row rta1_box_dan hta1_box_col hta1_box_row  \
0          08열          33연          08단          03열          25연   
1          08열          33연          08단          03열          25연   
2          08열          33연          08단          03열          25연   
3          08열          33연          08단          03열          25연   
4          08열          33연          08단          03열          25연   

  hta1_box_dan rta2_box_col rta2_box_row rta2_box_dan  ocv1_ocv ocv1_box_col  \
0          03단          04열          64연          06단      3333          22열   
1          03단          04열          64연          06단      3329          22열   
2          03단          04열          64연          06단      3329          22열   
3          03단          04열          64연          06단      3330          22열   
4          03단          04열          64연          06단      3330          22열   

  ocv1_box_dan  c1_curr_end  c1_voltage_avg  c1_capa  c1_ccval  c1_time_cc  \
0          08단         6801            3781     6451      6451        3417   
1          08단         6797            3783     6390      6390        3385   
2          08단         6798            3781     6436      6436        3410   
3          08단         6799            3781     6474      6474        3430   
4          08단         6799            3780     6486      6486        3436   

  c1_box_col c1_box_dan  c1_temp_avg  dc1_curr_end  dc1_voltage_avg  dc1_capa  \
0        22열        08단          268          6799             4108       544   
1        22열        08단          268          6800             4103       545   
2        22열        08단          268          6799             4107       544   
3        22열        08단          268          6800             4108       545   
4        22열        08단          268          6798             4110       544   

  dc1_box_col dc1_box_dan  dc1_temp_avg  dc1_capafit  c2_curr_end  \
0         22열         08단           266          539         6800   
1         22열         08단           266          540         6802   
2         22열         08단           266          539         6801   
3         22열         08단           266          540         6801   
4         22열         08단           266          539         6800   

   c2_voltage_avg  c2_capa  c2_ccval  c2_time_cc c2_box_col c2_box_dan  \
0            4146      625       625         332        22열        08단   
1            4144      657       657         349        22열        08단   
2            4146      630       630         335        22열        08단   
3            4146      627       627         333        22열        08단   
4            4147      613       613         326        22열        08단   

   c2_temp_avg  dc2_curr_end  dc2_voltage_avg  dc2_capa dc2_box_col  \
0          265          6799             4112       544         22열   
1          265          6800             4110       544         22열   
2          265          6799             4111       544         22열   
3          265          6799             4111       544         22열   
4          265          6799             4112       544         22열   

  dc2_box_dan  dc2_temp_avg  dc2_capafit  c3_curr_end  c3_voltage_avg  \
0         08단           266          539          135            4189   
1         08단           266          539          136            4189   
2         08단           266          539          136            4189   
3         08단           266          539          136            4190   
4      

In [164]:
df_raw_tat = pd.read_csv('/content/mydrive/MyDrive/posco-big-data/bat_tat.csv', encoding = 'euc-kr')


print(df_raw_tat.shape)
df_raw_tat.head()

(43754, 25)


/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (20,21,22,23,24) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


lot_id    tray_id                dt_start     cell_id  rta1_tat  hta1_tat  \
0  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10747    217066     86461   
1  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10748    217066     86461   
2  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10749    217066     86461   
3  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10750    217066     86461   
4  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10751    217066     86461   

   rta2_tat  ocv1_tat  c1_tat  dc1_tat  c2_tat  dc2_tat  c3_tat  dc3_tat  \
0      5732       119    3556      404     466      404    1797     3884   
1      5732       119    3556      404     466      404    1797     3884   
2      5732       118    3556      405     466      404    1797     3884   
3      5732       118    3556      405     466      404    1797     3884   
4      5732       119    3556      404     466      404    1797     3884   

   c4_tat  ocv2_tat  pg1_tat  pc1_tat  sa1_tat  socv1_tat sa2_tat socv2_tat  \
0    3428       118      290      290    53500          5  355357         5   
1    3428       118      290      290    53500          5  355357         5   
2    3428       118      290      290    53500          5  355357         5   
3    3428       118      290      290    53501          5  355357         5   
4    3428       118      290      290    53500          5  355357         5   

  sa3_tat socv3_tat sa4_tat  
0  215150         5  207532  
1  215150         5  207532  
2  215150         5  207532  
3  215150         5  207532  
4  215150         5  207532

In [ ]:
# df_raw.loc[df_raw['judge'] == '불량', 'judge'] = 1
# df_raw.loc[df_raw['judge'] == '양품', 'judge'] = 0

In [165]:
df_raw = pd.merge(df_raw_process, df_raw_tat)

print(df_raw.shape)
df_raw.head()

(43754, 130)


lot_id    tray_id                dt_start     cell_id judge  rta1_cell_no  \
0  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10747    양품            10   
1  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10748    양품             7   
2  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10749    양품            19   
3  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10750    양품            24   
4  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10751    양품            17   

  rta1_box_col rta1_box_row rta1_box_dan hta1_box_col hta1_box_row  \
0          08열          33연          08단          03열          25연   
1          08열          33연          08단          03열          25연   
2          08열          33연          08단          03열          25연   
3          08열          33연          08단          03열          25연   
4          08열          33연          08단          03열          25연   

  hta1_box_dan rta2_box_col rta2_box_row rta2_box_dan  ocv1_ocv ocv1_box_col  \
0          03단          04열          64연          06단      3333          22열   
1          03단          04열          64연          06단      3329          22열   
2          03단          04열          64연          06단      3329          22열   
3          03단          04열          64연          06단      3330          22열   
4          03단          04열          64연          06단      3330          22열   

  ocv1_box_dan  c1_curr_end  c1_voltage_avg  c1_capa  c1_ccval  c1_time_cc  \
0          08단         6801            3781     6451      6451        3417   
1          08단         6797            3783     6390      6390        3385   
2          08단         6798            3781     6436      6436        3410   
3          08단         6799            3781     6474      6474        3430   
4          08단         6799            3780     6486      6486        3436   

  c1_box_col c1_box_dan  c1_temp_avg  dc1_curr_end  dc1_voltage_avg  dc1_capa  \
0        22열        08단          268          6799             4108       544   
1        22열        08단          268          6800             4103       545   
2        22열        08단          268          6799             4107       544   
3        22열        08단          268          6800             4108       545   
4        22열        08단          268          6798             4110       544   

  dc1_box_col dc1_box_dan  dc1_temp_avg  dc1_capafit  c2_curr_end  \
0         22열         08단           266          539         6800   
1         22열         08단           266          540         6802   
2         22열         08단           266          539         6801   
3         22열         08단           266          540         6801   
4         22열         08단           266          539         6800   

   c2_voltage_avg  c2_capa  c2_ccval  c2_time_cc c2_box_col c2_box_dan  \
0            4146      625       625         332        22열        08단   
1            4144      657       657         349        22열        08단   
2            4146      630       630         335        22열        08단   
3            4146      627       627         333        22열        08단   
4            4147      613       613         326        22열        08단   

   c2_temp_avg  dc2_curr_end  dc2_voltage_avg  dc2_capa dc2_box_col  \
0          265          6799             4112       544         22열   
1          265          6800             4110       544         22열   
2          265          6799             4111       544         22열   
3          265          6799             4111       544         22열   
4          265          6799             4112       544         22열   

  dc2_box_dan  dc2_temp_avg  dc2_capafit  c3_curr_end  c3_voltage_avg  \
0         08단           266          539          135            4189   
1         08단           266          539          136            4189   
2         08단           266          539          136            4189   
3         08단           266          539          136            4190   
4      

In [166]:
df_raw.replace('.', np.NaN, inplace=True)
df_raw.replace(' ', np.NaN, inplace=True)

In [167]:
df_raw.dropna(inplace = True)

In [168]:
df_raw.shape

(43657, 130)

### **(21열, 22열)이랑 (6열, 9열, 18열) 나누기**

In [169]:
# 22열, 21열

df_raw22 = df_raw[df_raw['c4_box_col'] == '22열']
df_raw21 = df_raw[df_raw['c4_box_col'] == '21열']

df_raw_top = pd.concat([df_raw22, df_raw21], axis = 0)

print(df_raw_top.shape)
df_raw_top.head()

(4988, 130)


lot_id    tray_id                dt_start     cell_id judge  rta1_cell_no  \
0  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10747    양품            10   
1  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10748    양품             7   
2  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10749    양품            19   
3  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10750    양품            24   
4  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10751    양품            17   

  rta1_box_col rta1_box_row rta1_box_dan hta1_box_col hta1_box_row  \
0          08열          33연          08단          03열          25연   
1          08열          33연          08단          03열          25연   
2          08열          33연          08단          03열          25연   
3          08열          33연          08단          03열          25연   
4          08열          33연          08단          03열          25연   

  hta1_box_dan rta2_box_col rta2_box_row rta2_box_dan  ocv1_ocv ocv1_box_col  \
0          03단          04열          64연          06단      3333          22열   
1          03단          04열          64연          06단      3329          22열   
2          03단          04열          64연          06단      3329          22열   
3          03단          04열          64연          06단      3330          22열   
4          03단          04열          64연          06단      3330          22열   

  ocv1_box_dan  c1_curr_end  c1_voltage_avg  c1_capa  c1_ccval  c1_time_cc  \
0          08단         6801            3781     6451      6451        3417   
1          08단         6797            3783     6390      6390        3385   
2          08단         6798            3781     6436      6436        3410   
3          08단         6799            3781     6474      6474        3430   
4          08단         6799            3780     6486      6486        3436   

  c1_box_col c1_box_dan  c1_temp_avg  dc1_curr_end  dc1_voltage_avg  dc1_capa  \
0        22열        08단          268          6799             4108       544   
1        22열        08단          268          6800             4103       545   
2        22열        08단          268          6799             4107       544   
3        22열        08단          268          6800             4108       545   
4        22열        08단          268          6798             4110       544   

  dc1_box_col dc1_box_dan  dc1_temp_avg  dc1_capafit  c2_curr_end  \
0         22열         08단           266          539         6800   
1         22열         08단           266          540         6802   
2         22열         08단           266          539         6801   
3         22열         08단           266          540         6801   
4         22열         08단           266          539         6800   

   c2_voltage_avg  c2_capa  c2_ccval  c2_time_cc c2_box_col c2_box_dan  \
0            4146      625       625         332        22열        08단   
1            4144      657       657         349        22열        08단   
2            4146      630       630         335        22열        08단   
3            4146      627       627         333        22열        08단   
4            4147      613       613         326        22열        08단   

   c2_temp_avg  dc2_curr_end  dc2_voltage_avg  dc2_capa dc2_box_col  \
0          265          6799             4112       544         22열   
1          265          6800             4110       544         22열   
2          265          6799             4111       544         22열   
3          265          6799             4111       544         22열   
4          265          6799             4112       544         22열   

  dc2_box_dan  dc2_temp_avg  dc2_capafit  c3_curr_end  c3_voltage_avg  \
0         08단           266          539          135            4189   
1         08단           266          539          136            4189   
2         08단           266          539          136            4189   
3         08단           266          539          136            4190   
4      

In [170]:
# 9열, 4열, 6열

df_raw9 = df_raw[df_raw['c4_box_col'] == '09열']
df_raw4 = df_raw[df_raw['c4_box_col'] == '04열']
df_raw6 = df_raw[df_raw['c4_box_col'] == '06열']

df_raw_bot = pd.concat([df_raw9, df_raw4], axis = 0)
df_raw_bot = pd.concat([df_raw_bot, df_raw6], axis = 0)

print(df_raw_bot.shape)
df_raw_bot.head()

(4876, 130)


lot_id    tray_id               dt_start     cell_id judge  \
396  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10175    양품   
397  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10176    양품   
398  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10177    양품   
399  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10178    양품   
400  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10179    양품   

     rta1_cell_no rta1_box_col rta1_box_row rta1_box_dan hta1_box_col  \
396             1          07열          05연          01단          01열   
397             7          07열          05연          01단          01열   
398            24          07열          05연          01단          01열   
399             5          07열          05연          01단          01열   
400            14          07열          05연          01단          01열   

    hta1_box_row hta1_box_dan rta2_box_col rta2_box_row rta2_box_dan  \
396          24연          06단          06열          09연          08단   
397          24연          06단          06열          09연          08단   
398          24연          06단          06열          09연          08단   
399          24연          06단          06열          09연          08단   
400          24연          06단          06열          09연          08단   

     ocv1_ocv ocv1_box_col ocv1_box_dan  c1_curr_end  c1_voltage_avg  c1_capa  \
396      3339          09열          07단         6801            3782     6469   
397      3338          09열          07단         6798            3782     6448   
398      3337          09열          07단         6798            3782     6445   
399      3337          09열          07단         6802            3782     6455   
400      3335          09열          07단         6802            3781     6483   

     c1_ccval  c1_time_cc c1_box_col c1_box_dan  c1_temp_avg  dc1_curr_end  \
396      6469        3426        09열        07단          280          6800   
397      6448        3416        09열        07단          280          6799   
398      6445        3415        09열        07단          280          6799   
399      6455        3419        09열        07단          280          6801   
400      6483        3434        09열        07단          280          6799   

     dc1_voltage_avg  dc1_capa dc1_box_col dc1_box_dan  dc1_temp_avg  \
396             4109       545         09열         07단           279   
397             4107       544         09열         07단           279   
398             4107       544         09열         07단           279   
399             4108       545         09열         07단           279   
400             4109       544         09열         07단           279   

     dc1_capafit  c2_curr_end  c2_voltage_avg  c2_capa  c2_ccval  c2_time_cc  \
396          536         6800            4147      627       627         333   
397          535         6801            4145      635       635         337   
398          535         6801            4146      630       630         335   
399          536         6796            4146      636       636         338   
400          535         6800            4146      628       628         334   

    c2_box_col c2_box_dan  c2_temp_avg  dc2_curr_end  dc2_voltage_avg  \
396        09열        07단          278          6799             4113   
397        09열        07단          278          6799             4112   
398        09열        07단          278          6799             4112   
399        09열        07단          278          6799             4112   
400        09열        07단          278          6799             4113   

     dc2_capa dc2_box_col dc2_box_dan  dc2_temp_avg  dc2_capafit  c3_curr_end  \
396       544         09열         07단           277          536          136   
397       544         09열         07단           277          536          135   
398       544         09열         07단           277          536          136   
399       544         09열         07단           277          536          136   
400       5

In [171]:
df_raw_total = pd.concat([df_raw_top, df_raw_bot], axis = 0)
df_raw_total = df_raw_total.reset_index()

print(df_raw_total.shape)
df_raw_total.tail()

(9864, 131)


index   lot_id    tray_id               dt_start     cell_id judge  \
9859  43428  LOT-144  TRAY-2773  12-17-2021 오후 2:36:45  CELL-53553    양품   
9860  43429  LOT-144  TRAY-2773  12-17-2021 오후 2:36:45  CELL-53554    양품   
9861  43430  LOT-144  TRAY-2773  12-17-2021 오후 2:36:45  CELL-53555    양품   
9862  43431  LOT-144  TRAY-2773  12-17-2021 오후 2:36:45  CELL-54255    양품   
9863  43432  LOT-144  TRAY-2773  12-17-2021 오후 2:36:45  CELL-54256    양품   

      rta1_cell_no rta1_box_col rta1_box_row rta1_box_dan hta1_box_col  \
9859            12          07열          58연          09단          01열   
9860             9          07열          58연          09단          01열   
9861            24          07열          58연          09단          01열   
9862             1          07열          58연          09단          01열   
9863             2          07열          58연          09단          01열   

     hta1_box_row hta1_box_dan rta2_box_col rta2_box_row rta2_box_dan  \
9859          22연          10단          07열          42연          04단   
9860          22연          10단          07열          42연          04단   
9861          22연          10단          07열          42연          04단   
9862          22연          10단          07열          42연          04단   
9863          22연          10단          07열          42연          04단   

      ocv1_ocv ocv1_box_col ocv1_box_dan  c1_curr_end  c1_voltage_avg  \
9859      3350          06열          11단         6798            3781   
9860      3352          06열          11단         6798            3780   
9861      3350          06열          11단         6800            3780   
9862      3325          06열          11단         6799            3780   
9863      3325          06열          11단         6799            3780   

      c1_capa  c1_ccval  c1_time_cc c1_box_col c1_box_dan  c1_temp_avg  \
9859     6493      6493        3440        06열        11단          287   
9860     6486      6486        3436        06열        11단          287   
9861     6506      6506        3447        06열        11단          287   
9862     6453      6453        3418        06열        11단          287   
9863     6447      6447        3415        06열        11단          287   

      dc1_curr_end  dc1_voltage_avg  dc1_capa dc1_box_col dc1_box_dan  \
9859          6800             4113       544         06열         11단   
9860          6800             4114       544         06열         11단   
9861          6800             4114       545         06열         11단   
9862          6800             4109       544         06열         11단   
9863          6800             4109       544         06열         11단   

      dc1_temp_avg  dc1_capafit  c2_curr_end  c2_voltage_avg  c2_capa  \
9859           290          532         6800            4148      601   
9860           290          532         6801            4148      603   
9861           290          533         6799            4149      595   
9862           290          532         6801            4146      629   
9863           290          532         6802            4146      629   

      c2_ccval  c2_time_cc c2_box_col c2_box_dan  c2_temp_avg  dc2_curr_end  \
9859       601         319        06열        11단          290          6800   
9860       603         321        06열        11단          290          6800   
9861       595         316        06열        11단          290          6800   
9862       629         334        06열        11단          290          6800   
9863       629         334        06열        11단          290          6799   

      dc2_voltage_avg  dc2_capa dc2_box_col dc2_box_dan  dc2_temp_avg  \
9859             4116       544         06열         11단           290   
9860             4116       544         06열         11단           290   
9861             4116       545         06열         11단           290   
9862             4114       545         06열         11단           290   
9863             4114       544         06열         11단        

In [172]:
df_raw_total.loc[df_raw_total['c4_box_col'] == '04열', 'c4_box_col'] = 0
df_raw_total.loc[df_raw_total['c4_box_col'] == '06열', 'c4_box_col'] = 0
df_raw_total.loc[df_raw_total['c4_box_col'] == '09열', 'c4_box_col'] = 0

df_raw_total.loc[df_raw_total['c4_box_col'] == '21열', 'c4_box_col'] = 1
df_raw_total.loc[df_raw_total['c4_box_col'] == '22열', 'c4_box_col'] = 1

In [173]:
df_raw_total['c4_box_col'].value_counts()

1    4988
0    4876
Name: c4_box_col, dtype: int64

In [174]:
df_raw_total['c4_box_col'] = df_raw_total['c4_box_col'].astype(int)

#### **카이제곱검정**
- 설비와 불량 여부 사이에 관련성이 있는지 검정


In [ ]:
ct = pd.crosstab(df_raw_total['c4_box_col'], df_raw_total['judge'])
ct

judge        불량    양품
c4_box_col           
0            46  4830
1           421  4567

In [ ]:
stat, pval, df, expected = stats.chi2_contingency(observed = ct)

print('***** 카이제곱검정 *****')
print('H0 : 설비에 따라 양품불량개수의 차이가 없다.')
print('H1 : 설비에 따라 양품불량개수의 차이가 있다.')
print('')
print('통계량 : ', stat.round(3))
print('pvalue : ', pval.round(3))

***** 카이제곱검정 *****
H0 : 설비에 따라 양품불량개수의 차이가 없다.
H1 : 설비에 따라 양품불량개수의 차이가 있다.

통계량 :  305.593
pvalue :  0.0


- 검정통계량 305.593에 대한 유의확률 = 0.0 < 0.05이므로 귀무가설을 기각한다.
- 즉, 설비에 따라 양품불량개수의 차이가 있다.

### **ANOVA 검정**


In [ ]:
# ANOVA 실행

print('***** ANOVA분석 *****')
print('H0 : 설비에 따라 온도에 차이가 없다.')
print('H1 : 설비에 따라 온도에 차이가 있다.')
print('')
model = ols('c4_temp_avg ~ C(c4_box_col)', df_raw_total).fit()
anova_lm(model)

***** ANOVA분석 *****
H0 : 설비에 따라 온도에 차이가 없다.
H1 : 설비에 따라 온도에 차이가 있다.



df        sum_sq        mean_sq            F  PR(>F)
C(c4_box_col)     1.0  7.271406e+05  727140.612131  3634.647688     0.0
Residual       9862.0  1.972973e+06     200.058073          NaN     NaN

- 검정통계량 F = 3634.647688에 대한 유의확률 0.0 < 0.05이므로 귀무가설을 기각한다.
- 설비에 따라 온도에 차이가 있다.

### **2 sample t-test**



In [ ]:
# 정규성 검정

statistic1, p1 = stats.shapiro(df_raw_total.loc[df_raw_total['judge'] == '불량', 'c4_temp_avg'])
print("----------- 불량에 대한 정규성 검정 -----------")
print('H0 : 정규성을 만족한다.')
print('H1 : 정규성을 만족하지 않는다.')
print('')
print('통계량 : ', statistic1)
print('pvalue : ', p1)
print('')

statistic2, p2 = stats.shapiro(df_raw_total.loc[df_raw_total['judge'] == '양품', 'c4_temp_avg'])
print("----------- 양품에 대한 정규성 검정 -----------")
print('H0 : 정규성을 만족한다.')
print('H1 : 정규성을 만족하지 않는다.')
print('')
print('통계량 : ', statistic2)
print('pvalue : ', p2)
print('')

----------- 불량에 대한 정규성 검정 -----------
H0 : 정규성을 만족한다.
H1 : 정규성을 만족하지 않는다.

통계량 :  0.8832783102989197
pvalue :  2.611841188904814e-18

----------- 양품에 대한 정규성 검정 -----------
H0 : 정규성을 만족한다.
H1 : 정규성을 만족하지 않는다.

통계량 :  0.9911037683486938
pvalue :  8.634443905938565e-24



/usr/local/lib/python3.7/dist-packages/scipy/stats/morestats.py:1760: UserWarning: p-value may not be accurate for N > 5000.
  warnings.warn("p-value may not be accurate for N > 5000.")


- 불량과 양품 개수에 따른 온도의 유의확률 0.000, 0.000 < 0.05이므로 귀무가설을 기각한다.
- 두 집단 모두 정규성을 만족하지 않으므로 맨휘트니 검정을 통해 t검정을 진행한다.

In [177]:
big = df_raw_total[df_raw_total['c4_box_col'] == 1]
big.head()

index   lot_id    tray_id                dt_start     cell_id judge  \
0      0  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10747    양품   
1      1  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10748    양품   
2      2  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10749    양품   
3      3  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10750    양품   
4      4  LOT-100  TRAY-1023  12-01-2021 오전 10:01:25  CELL-10751    양품   

   rta1_cell_no rta1_box_col rta1_box_row rta1_box_dan hta1_box_col  \
0            10          08열          33연          08단          03열   
1             7          08열          33연          08단          03열   
2            19          08열          33연          08단          03열   
3            24          08열          33연          08단          03열   
4            17          08열          33연          08단          03열   

  hta1_box_row hta1_box_dan rta2_box_col rta2_box_row rta2_box_dan  ocv1_ocv  \
0          25연          03단          04열          64연          06단      3333   
1          25연          03단          04열          64연          06단      3329   
2          25연          03단          04열          64연          06단      3329   
3          25연          03단          04열          64연          06단      3330   
4          25연          03단          04열          64연          06단      3330   

  ocv1_box_col ocv1_box_dan  c1_curr_end  c1_voltage_avg  c1_capa  c1_ccval  \
0          22열          08단         6801            3781     6451      6451   
1          22열          08단         6797            3783     6390      6390   
2          22열          08단         6798            3781     6436      6436   
3          22열          08단         6799            3781     6474      6474   
4          22열          08단         6799            3780     6486      6486   

   c1_time_cc c1_box_col c1_box_dan  c1_temp_avg  dc1_curr_end  \
0        3417        22열        08단          268          6799   
1        3385        22열        08단          268          6800   
2        3410        22열        08단          268          6799   
3        3430        22열        08단          268          6800   
4        3436        22열        08단          268          6798   

   dc1_voltage_avg  dc1_capa dc1_box_col dc1_box_dan  dc1_temp_avg  \
0             4108       544         22열         08단           266   
1             4103       545         22열         08단           266   
2             4107       544         22열         08단           266   
3             4108       545         22열         08단           266   
4             4110       544         22열         08단           266   

   dc1_capafit  c2_curr_end  c2_voltage_avg  c2_capa  c2_ccval  c2_time_cc  \
0          539         6800            4146      625       625         332   
1          540         6802            4144      657       657         349   
2          539         6801            4146      630       630         335   
3          540         6801            4146      627       627         333   
4          539         6800            4147      613       613         326   

  c2_box_col c2_box_dan  c2_temp_avg  dc2_curr_end  dc2_voltage_avg  dc2_capa  \
0        22열        08단          265          6799             4112       544   
1        22열        08단          265          6800             4110       544   
2        22열        08단          265          6799             4111       544   
3        22열        08단          265          6799             4111       544   
4        22열        08단          265          6799             4112       544   

  dc2_box_col dc2_box_dan  dc2_temp_avg  dc2_capafit  c3_curr_end  \
0         22열         08단           266          539          135   
1         22열         08단           266          539          136   
2         22열         08단           266          539          136   
3         22열         08단           266          539          136   
4         22열         08단           266          539    

In [178]:
small = df_raw_total[df_raw_total['c4_box_col'] == 0]
small.head()

index   lot_id    tray_id               dt_start     cell_id judge  \
4988    396  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10175    양품   
4989    397  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10176    양품   
4990    398  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10177    양품   
4991    399  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10178    양품   
4992    400  LOT-100  TRAY-1008  12-01-2021 오전 6:36:25  CELL-10179    양품   

      rta1_cell_no rta1_box_col rta1_box_row rta1_box_dan hta1_box_col  \
4988             1          07열          05연          01단          01열   
4989             7          07열          05연          01단          01열   
4990            24          07열          05연          01단          01열   
4991             5          07열          05연          01단          01열   
4992            14          07열          05연          01단          01열   

     hta1_box_row hta1_box_dan rta2_box_col rta2_box_row rta2_box_dan  \
4988          24연          06단          06열          09연          08단   
4989          24연          06단          06열          09연          08단   
4990          24연          06단          06열          09연          08단   
4991          24연          06단          06열          09연          08단   
4992          24연          06단          06열          09연          08단   

      ocv1_ocv ocv1_box_col ocv1_box_dan  c1_curr_end  c1_voltage_avg  \
4988      3339          09열          07단         6801            3782   
4989      3338          09열          07단         6798            3782   
4990      3337          09열          07단         6798            3782   
4991      3337          09열          07단         6802            3782   
4992      3335          09열          07단         6802            3781   

      c1_capa  c1_ccval  c1_time_cc c1_box_col c1_box_dan  c1_temp_avg  \
4988     6469      6469        3426        09열        07단          280   
4989     6448      6448        3416        09열        07단          280   
4990     6445      6445        3415        09열        07단          280   
4991     6455      6455        3419        09열        07단          280   
4992     6483      6483        3434        09열        07단          280   

      dc1_curr_end  dc1_voltage_avg  dc1_capa dc1_box_col dc1_box_dan  \
4988          6800             4109       545         09열         07단   
4989          6799             4107       544         09열         07단   
4990          6799             4107       544         09열         07단   
4991          6801             4108       545         09열         07단   
4992          6799             4109       544         09열         07단   

      dc1_temp_avg  dc1_capafit  c2_curr_end  c2_voltage_avg  c2_capa  \
4988           279          536         6800            4147      627   
4989           279          535         6801            4145      635   
4990           279          535         6801            4146      630   
4991           279          536         6796            4146      636   
4992           279          535         6800            4146      628   

      c2_ccval  c2_time_cc c2_box_col c2_box_dan  c2_temp_avg  dc2_curr_end  \
4988       627         333        09열        07단          278          6799   
4989       635         337        09열        07단          278          6799   
4990       630         335        09열        07단          278          6799   
4991       636         338        09열        07단          278          6799   
4992       628         334        09열        07단          278          6799   

      dc2_voltage_avg  dc2_capa dc2_box_col dc2_box_dan  dc2_temp_avg  \
4988             4113       544         09열         07단           277   
4989             4112       544         09열         07단           277   
4990             4112       544         09열         07단           277   
4991             4112       544         09열         07단           277   
4992             4113       544         09열         07단        

In [181]:
# 맨휘트니 검정 (비모수적 t검정)
# 불량 여부에 따라 온도에 차이가 있는지

stat, p = stats.mannwhitneyu(big['c4_voltage_avg'], small['c4_voltage_avg'])

print('stat : {0:0.5f}'.format(stat))
print('p-value : {0:0.5f}'.format(p))

stat : 14539111.50000
p-value : 0.00000


- 불량 개수가 많은 집단과 적은 집단의 온도 차이에 대한 유의확률 0.000 < 0.05이므로 귀무가설을 기각한다.
- 양품과 불량 사이에 온도 차이가 존재한다.

In [ ]:
# 맨휘트니 검정 (비모수적 t검정)
# 불량 여부에 따라 전압에 차이가 있는지

stat, p = stats.mannwhitneyu(big['c4_temp_avg'], small['c4_temp_avg'])

print('stat : {0:0.5f}'.format(stat))
print('p-value : {0:0.5f}'.format(p))

- 불량 개수가 많은 집단과 적은 집단의 전압 차이에 대한 유의확률 0.000 < 0.05이므로 귀무가설을 기각한다.
- 양품과 불량 사이에 전압 차이가 존재한다.

In [182]:
# 맨휘트니 검정 (비모수적 t검정)
# 불량 여부에 따라 전류에 차이가 있는지

stat, p = stats.mannwhitneyu(big['c4_curr_end'], small['c4_curr_end'])

print('stat : {0:0.5f}'.format(stat))
print('p-value : {0:0.5f}'.format(p))

stat : 12396916.00000
p-value : 0.02905


- 불량 개수가 많은 집단과 적은 집단의 전류 차이에 대한 유의확률 0.029 < 0.05이므로 귀무가설을 기각한다.
- 양품과 불량 사이에 전류 차이가 존재한다.

**가설검정 결론**

- 중요한 공정조건 : 온도, 전압, 전류